File: go/find-first-and-last-position-of-element-in-sorted-array.go
func searchRange(nums []int, target int) []int {
	l, r := 0, len(nums)-1
	first, last := -1, -1

	for l <= r {
		m := (l + r) / 2
		if nums[m] >= target {
			first, r = m, m-1
		} else {
			l = m + 1
		}
	}

	if first == -1 || nums[first] != target {
		return []int{-1, -1}
	}

	l, r = 0, len(nums)-1
	for l <= r {
		m := (l + r) / 2
		if nums[m] <= target {
			last, l = m, m+1
		} else {
			r = m - 1
		}
	}

	return []int{first, last}
}

File: java/find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int first = -1, last = -1;
        int l = 0, r = nums.length-1;
        
        while (l <= r) {
            int m = (l+r)/2;
            if (nums[m] >= target) {
                first = m; r = m-1;
            } else {
                l = m+1;
            }
        }
        
        if (first == -1 || nums[first] != target) return new int[]{ -1, -1 };
        
        l = 0; r = nums.length-1;
        while (l <= r) {
            int m = (l+r)/2;
            if (nums[m] <= target) {
                last = m; l = m+1;
            } else {
                r = m-1;
            }
        }
        
        return new int[] { first, last };
    }
}